src/core/message-reply.ts

```diff
@@ -6,21 +6,21 @@ type Status = {
   detail: string
 };
 
-export type BaseMessageReply = {
+export type GenericMessageReply = {
   status: Status;
 };
 
-export function messageReplyFromError(e: unknown, code: number): BaseMessageReply {
+export function messageReplyFromError(e: unknown, code: number): GenericMessageReply {
 
   const detail = e instanceof Error ? e.message : 'Error';
 
   return { status: { code, detail } };
 }
 
 /**
- * Catch-all message reply type. It is recommended to use BaseMessageReply or a message-specific reply type whereever possible.
+ * Catch-all message reply type. It is recommended to use GenericMessageReply or a message-specific reply type whereever possible.
  */
-export type GenericMessageReply = BaseMessageReply & {
+export type UnionMessageReply = GenericMessageReply & {
```

Comment:
I wish I had removed `UnionMessageReply` altogether, though I'm not sure if it's possible while keeping `processMessage` in its current form. This type feels messy. If we've gotta keep it, I like this naming.

Reply:
Also, in `index.ts`, we're doing `export { UnionMessageReply as MessageReply }`. We should update to just `export { UnionMessageReply }`